mesonbuild/linkers/linkers.py

```diff
@@ -208,8 +208,12 @@ def can_linker_accept_rsp(self) -> bool:
         return self.can_rsp
 
     def get_std_link_args(self, env: 'Environment', is_thin: bool) -> T.List[str]:
-        # FIXME: osx ld rejects this: "file built for unknown-unsupported file format"
-        if is_thin and not env.machines[self.for_machine].is_darwin():
+        # Thin archives are a GNU extension not supported by the system linkers
+        # on Mac OS X, Solaris, or illumos, so don't build them on those OSes.
+        # osx ld rejects with: "file built for unknown-unsupported file format"
+        # illumos/solaris ld rejects with: "unknown file type"
```

Comment:
Nitpick, but stylize the names of the operating systems the same way. Mac OS X vs osx or Solaris vs solaris.

Reply:
Done